### DISCUSSION TITLE
Can RefinementList searchForItems functionality keep search state after refinement?

### DISCUSSION THREAD

**Discussion by ivark16** [algolia/instantsearch#5299] [Q&A]
Hi, so not sure if this should be a feature request but I'm having some trouble using the searchForItems in the RefinementList. I'm building an app using react-instantsearch-native package that heavily relies on algolia searching and filtering. What I want to be able to do is search for items in a filter (RefinementList) using searchForItems but when I refine the item I still want to be in the search as I was before. Which is not the current behaviour as seen in this [live code example](https://react-instantsearch.netlify.app/storybook/?path=/story/refinementlist--with-search-inside-items). The current behaviour is that when the item is refined it will again show the original refinementList with all the items. 

I have tried couple of work arounds to get desired result but still haven't succeeded e.g. 
- changing refinementList connector but algolia always returns `searchForFacetValuesResults` as `undefined` after refining
- calling `searchForItems` again after refinement with same search state but that does not work since the refinement has to be finished and since it is not a promise there is no way knowing when it has finished. 

My expected behaviour of this feature is that there would be some boolean value that can be triggered so the connector keeps the original search state after refining. 

Is there some workaround that I haven't tried/thought of? All suggestions/pointers appreciated 😄

**Comment by Haroenv** [comment]
If I understand it correctly, you want to refine while keeping the same items visible. Is that so you can select multiple items?

**Reply by ivark16** [reply]
Hi, yes it is so I can maybe select multiple items from the same search. Also I feel as it is now it is not straight forward that something was refined/selected as the filter is in another page in the app, so when clicking an item the user may not longer see it in the list as refined since it comes back to the user and showing the items at the top.

**Reply by Haroenv** [reply]
I see the use case when you have many facets that have similar names, but I'm not sure how it's possible to create that using the current design of the refinement list unfortunately.

With a custom implementation, you can force the search for facet values to be reapplied whenever the non-sffv items are shown: https://codesandbox.io/s/awesome-babbage-0j5fd?file=/src/App.js

**Reply by ivark16** [reply]
That's too bad, I think returning the same `searchForFacetValuesResults` in the connector after refinement is the only clean way to solve this. 

Regarding this custom implementation, I have already tried similar approach but I always see the original list for a split second since items come back as the original list after refinement. But I found another solution, by keeping the items in a state and only updating it when needed.

**Reply by sergiobarbosa360** [reply]
With this solution, by keeping the items on state, how you check if you need to update de items? You check if the searchState has changed?
